### DISCUSSION TITLE
Can't select multiple columns in bar / line chart

### DISCUSSION THREAD

**Discussion by shinny-yangyang** [perspective-dev/perspective#1747] [General]
### Steps to Reproduce:
 1. In official demo at homepage of https://perspective.finos.org/
 2. Switch to "X Bar" or "Y Line" mode.
 3. Select "bid" column in the column selection area.

### Expected Result:
"chg" and "bid" are both selected, and the chart shows both "chg" and "bid" as two series.

### Actual Result:
"chg" become unselected. the chart shows only "bid", 

### Environment:
Chrome and Edge browser

**Comment by texodus** [comment]
Thanks for the report @shinny-yangyang!  This is expected behavior (TL;DR - use `shift+click`), in that it was intended to work this way (explanation below).  However you're not the first person to bring this up as confusing, and I'm open to amending the default if there is a strong consensus to do so.

`<perspective-viewer>` has multiple selection modes that change based on the selected plugin.  For Datagrid, this mode is `toggle` - selecting columns via the checkbox icon adds them to the active column set.  For charts, this mode is `select`, which means choosing a columns makes that column the exclusive selected column (e.g., all other columns are deselected.

There are two easy ways to select additional columns -
1. Drag-and-drop the additional columns, they will append instead of replace.
2. Hold down the `shift` key while clicking will reverse the seleciton mode - in a Chart, `shift+click` will append the column, and in Datagrid `shift_click` will exclusively-select the column (the latter is also useful in Datagrid when you want to deselect all columns at once).

I'm going to move this to a discussion as its not a bug per se, but I'm curious if there is strong consensus for changing this default for e.g. -Line, -Bar, -Area single-axis chart types.

**Comment by jkusa** [comment]
I had similar expectations as @shinny-yangyang when first playing with perspective. I later stumbled upon the drag-and-drop technique for adding multiple measure/metric columns to the charts. Additionally, we've had a few stakeholders ask how to add multiple metrics series, as it was not initially intuitive. 

IMHO, I think changing the chart behavior to match the datagrid behavior would be a more intuitive and consistent experience for new users of perspective. Especially with the use of checkbox icons that give the sense that multiple selection will happen on click.

I actually was unaware of the shift-click behavior. If this change goes forward, I would be in favor for a shift-click to perform a exclusive column selection in the charts as done in the datagrid.

**Comment by jkusa** [comment]
@texodus any new thoughts or updates on this?
